app/controllers/paused_registration_controller.rb

```diff
@@ -139,13 +141,28 @@ def get_translated_service_name(simple_id)
     CONFIG_PROXY.get_transaction_translations(simple_id, params[:locale]).fetch(:name, nil)
   end
 
-  def get_idp_list(transaction_id)
+  def get_idp_list_for_registration(transaction_id)
     list = CONFIG_PROXY.get_available_idp_list_for_registration(transaction_id, "LEVEL_2")
     return nil if list.nil?
 
     list.idps
   end
 
+  def get_idp_list_for_sign_in
+    list = CONFIG_PROXY.get_idp_list_for_sign_in(last_rp)
+    return nil if list.nil?
+
+    list.idps
+  end
+
+  def last_idp_from_those_offering_sign_in
```

Comment:
```suggestion
  def last_idp_offers_sign_in
```
Make method name more clear

Reply:
That's not really what the method is doing though. That name change implies that a boolean is going to be returned. It's actually returning the IDP.

Reply:
But the current name implies that it return a boolean too. I think this logic could be split in two to make it easier to follow. We're doing two things - checking that last IDP offers sign in or registration, and then retrieving that IDP

Reply:
I don't feel the name does imply that. `last_idp` is a construct used throughout this controller, and it's being selected from those offering sign in.

Either way - i'll rewrite this logic to make it clearer.